test/node-api.js

```diff
@@ -6,11 +6,188 @@ chai.use(codeEquality);
 
 const { expect } = chai;
 
-const p = new Preprocessor();
+describe("parse", function() {
+  it("basic example", function() {
+    let p = new Preprocessor();
+    let output = p.parse("<template>Hello!</template>");
 
-describe("something", function () {
-  it("works for a basic example", function () {
-    let output = p.process("<template>Hi</template>");
+    expect(output).to.eql([
+      {
+        type: "template-tag",
+        tagName: "template",
+        contents: "Hello!",
+        range: {
+          start: 0,
+          end: 27
+        },
+        contentRange: {
+          start: 10,
+          end: 16
+        },
+        startRange: {
+          end: 10,
+          start: 0
+        },
+        endRange: {
+          start: 16,
+          end: 27
+        }
+      },
+    ]);
+  });
+
+  it("expression position", function() {
+    let p = new Preprocessor();
+    let output = p.parse("const tpl = <template>Hello!</template>");
+
+    expect(output).to.eql([
+      {
+        type: "template-tag",
+        tagName: "template",
+        contents: "Hello!",
+        range: {
+          start: 12,
+          end: 39
+        },
+        contentRange: {
+          start: 22,
+          end: 28
+        },
+        startRange: {
+          start: 12,
+          end: 22
+        },
+        endRange: {
+          start: 28,
+          end: 39
+        }
+      },
+    ]);
+  });
+
+  it("inside class body", function() {
+    let p = new Preprocessor();
+    let output = p.parse(`
+      class A {
+        <template>Hello!</template>
+      }
+    `);
+
+    expect(output).to.eql([
+      {
+        type: "template-tag",
```

Comment:
I think during office hours today, we talked about using `type` to differentiate between class body and non.
I don't remember what those terms were though.

ah-ha: https://github.com/emberjs/core-notes/pull/542/files#diff-db4a8200af245b39dd40e98deb07f9c80cbbdaa70dfbf8fa2ae0688f8e2f4634R35

here